crates/matrix-sdk/src/widget/client/implementation/mod.rs

```diff
@@ -0,0 +1,18 @@
+// Copyright 2023 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! High-level internal implementation of the client widget API logic.
+
+mod openid;
+mod outgoing;
```

Comment:
Why is this a separate module, instead of `openid` and `outgoing` being children of the `client` module directly?

Reply:
My rationale was that currently direct submodules of `client` are actually exposed to the parent module. The code in the `implementation` module is however not public and never exposed beyond `client` (i.e. it's only internally used by a client).

I thought it might be easier to digest the API when reading it (see different "layers" / hierarchies that interact with each other). But I don't have a strong opinion and could move them to the `client` module (I believe `client` may end up having way too many direct submodules, each of which corresponds to the different layers (low or high level) of the API that may make it a bit confusing. No strong opinion though :)

Reply:
I'd prefer not to nest as much for now. Besides, the other modules are *not* exposed to the parent module of `client`, just specific items from them are, via `pub(crate) use self::…`.

Reply:
Yes, I just thought that when we have a separate module for that, it's logically clear that "this whole part" is not exposed once you see that it's not imported, whereas if it's one in a single folder, then certain modules (`events.rs`) are simple and imported in `mod.rs` (they are there just to split a large `mod.rs`), whereas some other modules (`outgoing.rs` etc) that are rather a high level and never exposed (only used for the internal implementation) are on the same hierarchically.

I.e. by having an internal `implementation`, I tried to keep the "folder hierarchy" symmetrical to the object-relation hierarchy (low vs high level, etc).

Ok, I don't have a preference for now (we can change it later if you change your mind), so I removed nesting as you suggested, ptal 🙂